### DISCUSSION TITLE
Warn when running npm in a yarn project

### DISCUSSION THREAD

**Discussion by WesleyAC** [npm/feedback#244] [CLI] [answered]
I ran `npm install --save /path/to/local/file` in a yarn project by accident, and it outputted a bunch of errors and messed up the project in a way that required a `rm -rf node_modules && yarn` to fix, without any real indication of what I'd done wrong.

It seems like npm should be able to detect this and refuse to run if it's in a yarn project, or at least require some sort of `--force`-esque flag — that would have saved me a lot of time and confusion.

**Comment by MylesBorins** [comment, marked as answer]
@WesleyAC which version of npm were you using? npm 7 actually supports yarn lock files!

**Reply by WesleyAC** [reply]
Ah, I'm on `6.14.8`, since that seems to be the latest one packaged with NixOS currently. Good to know that this is now supported!
